omniledger/service/contracts.go

```diff
@@ -271,8 +271,9 @@ func (s *Service) ContractDarc(coll CollectionView, inst Instruction, coins []Co
 			if err != nil {
 				return nil, nil, errors.New("given darc could not be decoded: " + err.Error())
 			}
+			id := d.GetBaseID()
 			return []StateChange{
-				NewStateChange(Create, InstanceID{d.GetBaseID(), SubID{}}, ContractDarcID, darcBuf),
+				NewStateChange(Create, InstanceIDFromSlice(id), ContractDarcID, darcBuf, id),
 			}, coins, nil
 		}
 		// TODO The code below will never get called because this
```

Comment:
I don't know where this comment comes from, but it's wrong. Whenever we want to spawn a new instance, we need to send a `Spawn` instruction to an existing instance. This will often be a `ContractDarc` instance, which then will have to forward the spawn instruction to the appropriate contract.

Reply:
I wrote the comment here some time ago, maybe it's fixed now. I basically put a panic here and none of the tests panic'd. My understanding is that in order for this contract to be executed, OL will look up `GetContractState`, which simply returns `inst.Spawn.ContractID`. THence, the first if statement in this function always gets called and returns.

Reply:
But for the coins we send a `Spawn` instruction to the `DarcContract`, no?